libs/validator/domain_name_validator.hpp

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright AltPlus Inc., Ltd. 2017 All Rights Reserved.
+ * http://en.altplus.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef VALIDATOR_DOMAIN_NAME_VALIDATOR_HPP
+#define VALIDATOR_DOMAIN_NAME_VALIDATOR_HPP
+
+#include <string>
+
+namespace validator {
+  /**
+   * Parse a string as a domain name as defined in RFC 1035.
+   */
+  class DomainNameValidator {
```

Comment:
One [thing](https://github.com/hyperledger/iroha/pull/694#discussion_r150551815) left. What's the point of the class? Imo that's fine to use just `validator::isValidDomainName(..)`